src/aws_encryption_sdk/keyring/raw_keyring.py

```diff
@@ -0,0 +1,221 @@
+
+@attr.s
+class RawAESKeyring(Keyring):
+    """Public class for Raw AES Keyring.
+
+    :param str key_namespace: String defining the keyring ID
+    :param bytes key_name: Key ID
+    :param wrapping_key: Encryption key with which to wrap plaintext data key
+    :type wrapping_key: WrappingKey
+    :param wrapping_algorithm: Wrapping Algorithm with which to wrap plaintext data key
+    :type wrapping_algorithm: WrappingAlgorithm
+    """
+
+    key_namespace = attr.ib(validator=attr.validators.instance_of(six.string_types))
+    key_name = attr.ib(hash=True, validator=attr.validators.instance_of(six.binary_type))
+    _wrapping_key = attr.ib(hash=True, repr=False, validator=attr.validators.instance_of(WrappingKey))
+    _wrapping_algorithm = attr.ib(repr=False, validator=attr.validators.instance_of(WrappingAlgorithm))
+
+    def __attrs_post_init__(self):
+
+        _key_provider = MasterKeyInfo(provider_id=self.key_namespace, key_info=self.key_name)
+
+        _key_info_prefix = struct.pack(
+            ">{}sII".format(len(self.key_name)),
+            to_bytes(self.key_name),
+            # Tag Length is stored in bits, not bytes
+            self._wrapping_algorithm.algorithm.tag_len * 8,
+            self._wrapping_algorithm.algorithm.iv_len,
+        )
+
+    def on_encrypt(self, encryption_materials):
+        """Generate a data key if not present and encrypt it using any available wrapping key.
+
+        :param encryption_materials: Contains signing key, encryption context and algorithm suite
+                                    required to encrypt data key
+        :type encryption_materials: aws_encryption_sdk.materials_managers.EncryptionMaterials
+        :returns: Contains signing key, unencrypted data key, encrypted data keys,
+                                    encryption context and algorithm suite required to encrypt data key
+        :rtype: aws_encryption_sdk.materials_managers.EncryptionMaterials
+        """
+
+        encryption_materials = on_encrypt_helper(
+            encryption_materials, self._key_provider, self._wrapping_key, self._wrapping_algorithm, self.key_name
+        )
+
+        return encryption_materials
+
+    def on_decrypt(self, decryption_materials, encrypted_data_keys):
+        """Attempt to decrypt the encrypted data keys.
+
+        :param decryption_materials: Contains verification key, algorithm, encryption context and keyring trace.
+        :type decryption_materials: aws_encryption_sdk.materials_managers.DecryptionMaterials
+        :param encrypted_data_keys: List of encrypted data keys.
+        :type: List of `aws_encryption_sdk.structures.EncryptedDataKey`
+        :returns: Contains verification key, algorithm, data_encryption_key, encryption context and keyring trac
+        :rtype: aws_encryption_sdk.materials_managers.DecryptionMaterials
+        """
+
+        # Decrypt data key
+        expected_key_info_len = len(self._key_info_prefix) + self._wrapping_algorithm.algorithm.iv_len
```

Comment:
`encrypted_data_keys` is an iterable of `EncryptedDataKey`. The logic in this method expects a single `EncryptedDataKey`.

Once you address that, you'll need to also add checks to stop iterating through the encrypted data keys once the data encryption key is set.

Ditto on `on_decrypt` in the other keyring.